pre-commit.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
+#
+# hook script to verify what is about to be committed.
+# Called by "git commit" with no arguments.  The hook should
+# exit with non-zero status after issuing an appropriate message if
+# it wants to stop the commit.
+#
+
+# Check 1: check for direct commit against protected branches.
+# ------------------------------------------------------------
+function validate_branch () {
+    # notice current_branch evaluation only looks at part at the end of
+    # so if you want to exclude feature/foo, list foo in the protected_branches
+
+    PROTECTED_BRANCHES=('master' 'develop' 'support-1.x')
```

Comment:
`develop` should not be a protected branch.
```suggestion
    PROTECTED_BRANCHES=('master' 'support-1.x')
```

Reply:
why isn't develop protected ? in what cases do we commit to it directly? if it's only very rare occurrences I recommend using `git commit --no-verify` than opening up  `develop` here.